src/partition/repart.c

```diff
@@ -375,6 +399,37 @@ static int context_add_free_area(
         return 0;
 }
 
+static Partition* find_verity_sibling(Partition *head, Partition *p) {
+        assert(p);
+        assert(p->verity != VERITY_OFF);
+        assert(p->verity_name);
+
+        /* Try to find the matching sibling partition for a verity partition. For a data partition, this is
+         * the corresponding hash partiton with the same verity name (and vice versa for the hash partition).
+         */
+
+        LIST_FOREACH(partitions, q, head) {
+                if (p == q)
+                        continue;
+
+                if (q->verity == VERITY_OFF)
+                        continue;
+
+                if (p->verity == q->verity)
+                        continue;
+
+                assert(q->verity_name);
+
+                if (!streq(p->verity_name, q->verity_name))
+                        continue;
+
+                return q;
+        }
+
+        return NULL;
+}
```

Comment:
needs some protection if people assign the same match key to multiuple partitions